predicators/planning.py

```diff
@@ -650,6 +650,158 @@ def _update_nsrts_with_failure(
     return new_predicates, new_ground_nsrts
 
 
+def _update_sas_file_with_failure(discovered_failure: _DiscoveredFailure,
+                                  sas_file: str) -> None:  # pragma: no cover
+    """Update the given sas_file of ground_nsrts for FD based on the given
+    DiscoveredFailure.
+
+    We directly update the sas_file with the new ground_nsrts.
+    """
+    # Get string representation of the ground_nsrt with the Discovered Failure.
+    ground_op_str = discovered_failure.failing_nsrt.name.lower(
+    ) + " " + " ".join(o.name for o in discovered_failure.failing_nsrt.objects)
+    # Add Discovered Failure for each offending object.
+    for obj in discovered_failure.env_failure.info["offending_objects"]:
+        with open(sas_file, 'r', encoding="utf-8") as f:
+            sas_lines = f.readlines()
+        # For every line in our sas_file we are going to copy it to our
+        # new_sas_file and make edits as needed.
+        new_sas_file = []
+        sas_file_i = 0
+
+        # Fix sas_file Variables:
+        # The first line is "begin_variable".
+        # The second line contains the name of the variable (which is
+        # usually a nondescriptive name like "var7").
+        # The third line specifies the axiom layer of the variable.
```

Comment:
The axiom layer is for adding axioms instead of single value variables in Fast Downward. Note: "For state variables that do not correspond to axioms, i.e. which are not computed from the values of other state variables, the axiom layer is always -1"
Link: https://www.fast-downward.org/TranslatorOutputFormat